File: src/main/java/com/innovativecore/patientappointment/common/event/patient/PatientRegisteredEvent.java
package com.innovativecore.patientappointment.common.event.patient;

import lombok.Value;

import java.time.Instant;
import java.time.LocalDate;

@Value
public class PatientRegisteredEvent{
    String patientId;
    String  firstName;
    String  lastName;
    LocalDate dateOfBirth;
    String gender;
    String  contactNumber;
    String  address;
    Instant registeredAt;


    }

File: src/main/java/com/innovativecore/patientappointment/common/event/PatientRegisteredEvent.kt
package com.innovativecore.patientappointment.common.event

import java.time.Instant
import java.time.LocalDate
import java.util.UUID


data class PatientRegisteredEvent(
    val patientId: UUID,
    val firstName: String,
    val lastName: String,
    val dateOfBirth: LocalDate,
    val gender: String,
    val contactNumber: String,
    val address: String,
    val registrationTimestamp: Instant
)
